L03-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,32 @@
+package homework;
+
+import java.util.AbstractMap;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomerService {
+    private final TreeMap<Customer, String> cus;
```

Comment:
```suggestion
    private final NavigableMap<Customer, String> cus;
```

Тип лучше объявлять минимальным интерфейсом. Здесь достаточно `NavigableMap`.